app/models/forms/BankDetailsForm.scala

```diff
@@ -16,64 +16,69 @@
 
 package models.forms
 
-import essttp.rootmodel.bank.{SortCode, AccountNumber}
+import essttp.rootmodel.bank.{AccountName, AccountNumber, SortCode}
 import models.enumsforforms.IsSoleSignatoryFormValue
 import play.api.data.Forms.{mapping, nonEmptyText}
 import play.api.data.validation.{Constraint, Invalid, Valid}
 import play.api.data.{Form, Forms, Mapping}
 import util.EnumFormatter
 
 final case class BankDetailsForm(
-    name:            String,
+    name:            AccountName,
     sortCode:        SortCode,
     accountNumber:   AccountNumber,
     isSoleSignatory: IsSoleSignatoryFormValue
 )
-object BankDetailsForm {
-  def form: Form[BankDetailsForm] = {
-
-    val sortCodeRegex: String = "^[0-9]{6}$"
-
-    val sortCodeConstraint: Constraint[SortCode] =
-      Constraint(sortCode =>
-        if (!sortCode.value.forall(_.isDigit)) Invalid("error.nonNumeric")
-        else if (sortCode.value.matches(sortCodeRegex)) Valid
-        else Invalid("error.invalid"))
-
-    val sortCodeMapping: Mapping[SortCode] = nonEmptyText
-      .transform[SortCode](
-        s => SortCode(s.replaceAllLiterally("-", "").replaceAll("\\s", "")),
-        _.value
-      ).verifying(sortCodeConstraint)
-
-    val accountNumberRegex: String = "^[0-9]{6,8}$"
-
-    val accountNumberConstraint: Constraint[AccountNumber] =
-      Constraint(accountNumber =>
-        if (!accountNumber.value.forall(_.isDigit)) Invalid("error.nonNumeric")
-        else if (accountNumber.value.matches(accountNumberRegex)) Valid
-        else Invalid("error.invalid"))
-
-    val accountNumberMapping: Mapping[AccountNumber] = nonEmptyText
-      .transform[AccountNumber](
-        s => AccountNumber(s.replaceAll("\\s", "")),
-        _.value
-      ).verifying(accountNumberConstraint)
-
-    val isSoleSignatoryFormMapping: Mapping[IsSoleSignatoryFormValue] = Forms.of(EnumFormatter.format(
-      enum                    = IsSoleSignatoryFormValue,
-      errorMessageIfMissing   = "error.required",
-      errorMessageIfEnumError = "error.required"
-    ))
 
+object BankDetailsForm {
+  def form: Form[BankDetailsForm] =
     Form(
       mapping(
-        "name" -> nonEmptyText(maxLength = 100),
+        "name" -> accountNameMapping,
         "sortCode" -> sortCodeMapping,
         "accountNumber" -> accountNumberMapping,
         "isSoleSignatory" -> isSoleSignatoryFormMapping
       )(BankDetailsForm.apply)(BankDetailsForm.unapply)
     )
 
+  val accountNameConstraint: Constraint[AccountName] = {
+    Constraint(accountName =>
+      if (accountName.value.length < 100) Valid
+      else Invalid("error.invalid"))
   }
+  val accountNameMapping: Mapping[AccountName] =
+    nonEmptyText.transform[AccountName](name => AccountName(name), _.value).verifying(accountNameConstraint)
+
+  val sortCodeRegex: String = "^[0-9]{6}$"
+  val sortCodeConstraint: Constraint[SortCode] =
```

Comment:
should be be stripping spaces for the sort code and account number? And maybe even the various different dashes/hyphens for sort code? That'd be more forgiving on the users which is encourages for accessibility.

Reply:
Can do, I just copied it as it was, as I moved it out of the def

Reply:
Oh sorry I think it does, in the sortCodeMapping:
`val sortCodeMapping: Mapping[SortCode] = nonEmptyText
    .transform[SortCode](
      s => SortCode(s.replaceAllLiterally("-", "").replaceAll("\\s", "")),
      _.value
    ).verifying(sortCodeConstraint)`

Reply:
ah ok, missed that. There's a bunch of other dashes that we could filter out as well like these ones:

https://github.com/hmrc/help-to-save-frontend/blob/main/app/uk/gov/hmrc/helptosavefrontend/forms/BankDetailsValidation.scala#L43

Reply:
I think we have a ticket to enhance the validation on the bank/direct debit details page, can we address this in that ticket?

Reply:
if it's going to be revisited later on then that makes sense yeah 👍

Reply:
https://jira.tools.tax.service.gov.uk/browse/SUPP-398
It was originally for just account name, but I've altered for all fields on the screen. It does need refining though